projects/flipper-components/src/lib/services/pouchdb.service.ts

```
@@ -62,6 +62,26 @@ export class PouchDBService {
         })
     
     }
+    public async fastQuery(fields = [],selector = {}){
+        try {
+            // Create the Index
+            var result = await this.database.createIndex({
+                index:  { fields: fields }
+            });
+        
+            // Query the Index using find().
+            result = await this.database.find({
+                
+                selector: selector
+            });
+        
+            // Found docs are in result.docs
+            return await result.docs;
+        }
+        catch( err ) {
+            console.log(err);
```

Can we return a little failure on UI so we notify a user about the issue I guess will be much more user friendly.